ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -274,28 +291,95 @@ private boolean applyMigration(
+  ) {
+    final List<SqlCommand> commands = CommandParser.parse(migrationFileContent);
+    for (final SqlCommand command : commands) {
       try {
-        ksqlClient.executeStatement(command).get();
+        executeCommand(command, ksqlClient, restClient);
       } catch (InterruptedException | ExecutionException e) {
         final String errorMsg = String.format(
-            "Failed to execute sql: %s. Error: %s", command, e.getMessage());
-        LOGGER.error(errorMsg);
+            "Failed to execute sql: %s. Error: %s", command.getCommand(), e.getMessage());
         updateState(config, ksqlClient, MigrationState.ERROR,
             executionStart, migration, clock, previous, Optional.of(errorMsg));
-        return false;
+        throw new MigrationException(errorMsg);
       }
     }
+  }
 
-    updateState(config, ksqlClient, MigrationState.MIGRATED,
-        executionStart, migration, clock, previous, Optional.empty());
-    LOGGER.info("Successfully migrated");
-    return true;
+  private void executeCommand(
+      final SqlCommand command,
+      final Client ksqlClient,
+      final KsqlRestClient restClient
+  ) throws ExecutionException, InterruptedException {
+    if (command instanceof SqlStatement) {
+      ksqlClient.executeStatement(command.getCommand()).get();
+    } else if (command instanceof SqlInsertValues) {
+      final List<FieldInfo> fields =
+          ksqlClient.describeSource(((SqlInsertValues) command).getSourceName()).get().fields();
+      ksqlClient.insertInto(
+          ((SqlInsertValues) command).getSourceName(),
+          getRow(
+              fields,
+              ((SqlInsertValues) command).getColumns(),
+              ((SqlInsertValues) command).getValues())).get();
+    } else if (command instanceof SqlConnectorStatement) {
+      restClient.makeKsqlRequest(command.getCommand());
+    }
+  }
+
+  private KsqlObject getRow(
+      final List<FieldInfo> sourceFields,
+      final List<String> insertColumns,
+      final List<Expression> insertValues
+  ) {
+    final Map<String, Object> row = new HashMap<>();
+    if (insertColumns.size() > 0) {
+      verifyColumnValuesMatch(insertColumns, insertValues);
+      for (int i = 0 ; i < insertColumns.size(); i++) {
+        row.put(insertColumns.get(i), CommandParser.toFieldType(insertValues.get(i)));
+      }
+    } else {
+      final List<String> columnNames = sourceFields.stream()
+          .map(FieldInfo::name).collect(Collectors.toList());
+      verifyColumnValuesMatch(columnNames, insertValues);
+      for (int i = 0 ; i < sourceFields.size(); i++) {
+        row.put(sourceFields.get(i).name(), CommandParser.toFieldType(insertValues.get(i)));
+      }
+    }
+
+    return new KsqlObject(row);
+  }
+
+  private void verifyColumnValuesMatch(final List<String> columns, final List<Expression> values) {
+    if (columns.size() != values.size()) {
+      throw new MigrationException("Expected number columns and values to match: "
```

Comment:
nit: reword the error message to make it clear that this is thrown due to user error (rather than something wrong with the migrations tool):
```suggestion
      throw new MigrationException("Invalid `INSERT VALUES` statement. Number of columns and values must match. Got: Columns: %d. Values: %d."
```